extensions/interactions/MathExpressionInput/MathExpressionInput.js

```diff
@@ -0,0 +1,112 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+/**
+ * Directive for the MathExpressionInput interaction.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+oppia.directive('oppiaInteractiveMathExpressionInput', [
+  'oppiaHtmlEscaper', 'mathExpressionInputRulesService',
+  function(oppiaHtmlEscaper, mathExpressionInputRulesService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: 'interaction/MathExpressionInput',
+      controller: [
+        '$scope', '$attrs', '$timeout', '$element',
+        'LABEL_FOR_CLEARING_FOCUS', function(
+            $scope, $attrs, $timeout, $element, LABEL_FOR_CLEARING_FOCUS) {
+
+          var guppyDivElt = $element[0].querySelector('.guppy-div');
+          var guppyInstance = new Guppy(guppyDivElt, {});
+          var guppyDivId = guppyInstance.editor.id;
+
+          var labelForFocusTarget = $attrs.labelForFocusTarget || null;
+
+          $scope.$on('focusOn', function(e, name) {
+            if (!labelForFocusTarget) {
+              return;
+            }
+
+            if (name === labelForFocusTarget) {
+              guppyInstance.activate();
+            } else if (name === LABEL_FOR_CLEARING_FOCUS) {
+              guppyInstance.deactivate();
+            }
+          });
+
+          guppyInstance.done_callback = function() {
+            submitAnswerForEvaluation();
+          };
+
+          var submitAnswerForEvaluation = function() {
+            var answer = {
+              ascii: Guppy.instances[guppyDivId].get_content('calc'),
+              latex: Guppy.instances[guppyDivId].get_content('latex')
+            };
+
+            if (answer !== undefined && answer !== null) {
+              $scope.$parent.submitAnswer(
+                answer, mathExpressionInputRulesService);
+            }
+          };
+
+          $scope.submitAnswer = submitAnswerForEvaluation;
```

Comment:
Is it necessary to define a new function, and then set $scope.submitAnswer to it? Can't you just define $scope.submitAnswer directly?